CMakeLists.txt

```
@@ -309,11 +308,12 @@ endif()
 
 # Don't use PCH w/GCC (https://bugzilla.redhat.com/show_bug.cgi?id=1721553#c34)
 if (NOT CMAKE_CXX_COMPILER_ID STREQUAL GNU)
-    target_precompile_headers(${PROJECT_NAME} PRIVATE lib/converters.h)
+    target_precompile_headers(${PROJECT_NAME} PRIVATE Quotient/converters.h)
 endif ()
 
 target_include_directories(${PROJECT_NAME} PUBLIC
-    $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/lib>
+    $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>
+    $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/Quotient>
```

I think this include path should also obey `QUOTIENT_FORCE_NAMESPACED_INCLUDES`, otherwise using libQuotient as a submodule is not coherent with using it as an external library (this is the case of Quaternion; it can be built both ways). Something like
```suggestion
    $<BUILD_INTERFACE:$<$<NOT:$<BOOL:${QUOTIENT_FORCE_NAMESPACED_INCLUDES}>>:${CMAKE_CURRENT_SOURCE_DIR}/Quotient>
```
That, in turn will cause the library to fail building because quite a few files use the fact that the topmost source directory is in the include path. This is not a good assumption anyway, and these files should be corrected - except quotest and autotests, perhaps. Ideally I'd like `QUOTIENT_FORCE_NAMESPACED_INCLUDES` not to apply to them (not to autotests for sure; quotest is arguable).

---

I've made libQuotient itself buildable with namespaced includes, I still have to do the buildsystem part in a way it doesn't affect the tests though.

---

This is what I got as a result: https://github.com/quotient-im/libQuotient/tree/namespaced-includes

---

...oh, alright, I'm just a bit too late :)

---

...and that being said - you still haven't applied this suggestion (i.e. actually enforcing `QUOTIENT_FORCE_NAMESPACED_INCLUDES` in `BUILD_INTERFACES`) - is it intentional?

---

That's what I meant with "I still have to do the buildsystem part" above :) (I'm at LAS at the moment and have to do this in the breaks).

---

... and should all be there now.